pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -377,6 +388,40 @@ func (p *provider) parseTopologySpreadConstraint(topologyConstraints []kubevirtt
 	return parsedTopologyConstraints, nil
 }
 
+func (p *provider) parseOSImage(primaryDisk kubevirttypes.PrimaryDisk, nameSpace string) (*OSImage, error) {
+	osImage, err := p.configVarResolver.GetConfigVarStringValue(primaryDisk.OsImage)
+	if err != nil {
+		return nil, fmt.Errorf(`failed to get value of "primaryDisk.osImage" field: %w`, err)
+	}
+
+	var parsedOSImage *OSImage
+	// If PrimaryDiskOSSource is specified
```

Comment:
I would extend this comment and say that if source is specified then that means we deal with new API